test/Test/Pos/Ssc/GodTossing/Toss/BaseSpec.hs

```diff
@@ -0,0 +1,573 @@
+    description_notVerifiesBadSig =
+        "unsuccessfully verifies a commitment signature and a mismatching epoch"
+    description_verifiesOkOpening =
+        "successfully verifies that an opening corresponding to the given commitment\
+\ does indeed belong to it"
+    description_notVerifiesBadOpening =
+        "unsuccessfully verifies a mismatching commitment and opening"
+    description_emptyPayload =
+        "an empty payload always leads to successful verification"
+    description_checksGoodCommsPayload =
+        "successfully checks payload of commitments with a proper epoch index"
+    description_checksBadCommsPayload =
+        "unsuccessfully checks payload of commitments with the right exception for each\
+        \ failure case"
+    description_checksGoodOpensPayload =
+        "successfully checks payload of openings when the global state is correct"
+    description_checksBadOpeningsPayload =
+        "unsuccessfully checks payload of openings with the right exception for each\
+        \ failure case"
+    description_checksGoodSharesPayload =
+        "successfully checks payload of shares"
+    description_checksBadSharesPayload =
+        "unsuccessfully checks payload of shares with the right exception for each\
+        \ failure case"
+    description_checksGoodCertsPayload =
+        "successfully checks valid payload of VSS certificates"
+    description_checksBadCertsPayload =
+        "unsuccessfully checks payload of certificates with the right exception for each\
+        \ failure case"
+
+verifiesOkComm :: CommitmentOpening -> Bool
+verifiesOkComm CommitmentOpening{..} =
+    verifyCommitment coCommitment
+
+notVerifiesBadComm :: BadCommitment -> Bool
+notVerifiesBadComm (getBadComm -> badComm) =
+    not . verifyCommitment $ badComm
+
+verifiesOkCommSig :: SecretKey -> Commitment -> EpochIndex -> Bool
+verifiesOkCommSig sk comm epoch =
+    let commSig = (toPublic sk, comm, sign SignCommitment sk (epoch, comm))
+    in verifyCommitmentSignature epoch commSig
+
+notVerifiesBadCommSig :: BadSignedCommitment -> EpochIndex -> Bool
+notVerifiesBadCommSig (getBadSignedC -> badSignedComm) epoch =
+    not $ verifyCommitmentSignature epoch badSignedComm
+
+verifiesOkOpening :: CommitmentOpening -> Bool
+verifiesOkOpening CommitmentOpening{..} =
+    verifyOpening coCommitment coOpening
+
+notVerifiesBadOpening :: BadCommAndOpening -> Bool
+notVerifiesBadOpening (getBadCAndO -> badCommsAndOp) =
+    not . uncurry verifyOpening $ badCommsAndOp
+
+emptyPayload
+    :: Monoid container
+    => (container -> ExceptT e PureToss a)
+    -> MultiRichmenStake
+    -> GtGlobalState
+    -> Bool
+emptyPayload pureToss mrs gtgs =
+    isRight $ tossRunner mrs gtgs $ pureToss mempty
+
+newtype GoodCommsPayload = GoodComms
+    { getGoodComms :: (EpochIndex, GtGlobalState, CommitmentsMap, MultiRichmenStake)
+    } deriving (Show, Eq)
+
+instance Arbitrary GoodCommsPayload where
```

Comment:
I was thinking about it before I wrote that comment and I found it questionable actually. On the one hand, you're right, these `Arbitrary` instances and helpers are used only in tests. On the other hand, I can imagine that some of these `Arbitrary` instances may be useful later to implement `Arbitrary` for some complex data type (which is used in other code, not some ad-hoc type for tests). And `Arbitrary` for this complex type may be needed not only in tests.

> try and differentiate between actual Arbitrary instances and test ones.

Ok, let's do it for now. Even if later we notice that we need some of these `Arbitrary` in places other than tests, we will move it, it shouldn't be a problem.